### PR TITLE
Fix gradient regressions

### PR DESCRIPTION
**Pull request by pmconne**
#4035 changed the behavior of `Gradient.Symb.getImage` in two ways:
- No longer constrains the width to 1 for thematic gradients; and
- No longer includes the margin color for thematic gradients.

A supposition that `getImage` is never used by the renderer was incorrect: it is used when producing texture images for gradients, e.g. in decoration graphics or  tiles. This produced image differences in the "analysis example" models (and the "analysis style example" in display-test-app): the margin color no longer showed at the extremes of these models. While in this particular case that looks like an improvement, the margin color is important for highlighting portions of the model outside of the thematic range.

This regression also highlighted a much older regression: prior to [this PR](https://github.com/iTwin/itwinjs-core/commit/4fcaa13a130ed458794a2a3d92dde743aa154920) we always made thematic gradient textures 8192 pixels tall, in an effort to minimize bleeding of the margin color. That PR changed it to use whatever height was passed in; in this case, the caller passes in 256 which is why the margin color shows up so pronounced in the analysis example models.

Moreover, while #4035 did not technically break the API it did break the behavior, which is arguably worse - at least with an API break you find out at compile time that something you depend on has changed.

Fixes:
- `Gradient.Symb.getImage` now behaves exactly as it did prior to #4035.
- Added `Gradient.Symb.produceImage` which takes width, height, and optional flag indicating thematic margin should be included. It does not constrain the width of thematic images - that should always have been left up to the caller.
- `System.getGradientTexture` now requests an image of size (1, maxTextureSize) for thematic gradients, to minimize margin color bleeding.

@eringram you will want to use `produceImage` instead of `getImage` for your sample/tutorial.
@markschlosseratbentley @calebmshafer ~~we need to backport this to 3.3 prior to release.~~ #4035 is not included in the 3.3 release branch.

### CODE REVIEW

**Comment by pmconne** [comment]
@Mergifyio backport release/3.3.x

File: common/api/summary/core-frontend.exports.csv [hunk 1]
```diff
@@ -441,6 +441,7 @@ public;ParticleProps
 internal;PatternGraphicParams
 internal;PerformanceMetrics
 public;PerModelCategoryVisibility
+beta;Props
```
[the remark this answers is not in the document]
Comment (pmconne) [reply common/api/summary/core-frontend.exports.csv:444]: You'll have to ask api-extractor about that. @nick4598 recently added `PerModelCategoryVisibility.Props` as `beta`.
If you can find the other PR that may provide a clue.

[the remark this answers is not in the document]
Comment (calebmshafer) [reply common/api/summary/core-frontend.exports.csv:444]: Okay, it may be that the exports weren't updated when api extractor ran.

**Review by calebmshafer — APPROVED** [review]

**Comment by pmconne** [comment]
@DStradley please review the image differences vs master for "pmc-fix-gradient-textures_2". Increasing the height of the image from 256 to (max texture size) almost entirely eliminates the bands of black margin color, and reduces the blending between different gradient colors producing more distinct bands. It also changes the hue of the static flat mesh in AnalysisExample_XSWS0_SolidFill_+ovrEl.png, but much more subtly than observed after @eringram's change.

These all look like improvements to me; if you agree, please update the reference images.

**Comment by DStradley** [comment]
> These all look like improvements to me; if you agree, please update the reference images.

Agreed and done.  Do we need to add a gradient test case for margins?

**Review by markschlosseratbentley — APPROVED** [review]

**Pull request merged**